precompiles/assets/assets.go

```diff
@@ -86,9 +85,9 @@ func (p Precompile) Run(evm *vm.EVM, contract *vm.Contract, readOnly bool) (bz [
 	// It avoids panics and returns the out of gas error so the EVM can continue gracefully.
 	defer cmn.HandleGasError(ctx, contract, initialGas, &err)()
 
-	if err := stateDB.Commit(); err != nil {
-		return nil, err
-	}
+	// we cache the context because we consume the failure, if any, in this function.
+	// cascading it to statedb is possible but we will lose foundry compatibility
+	// see https://github.com/imua-xyz/imuachain/issues/70
 	cc, writeFunc := ctx.CacheContext()
```

Comment:
both evmos's patch and cronos's implementation seems not having this, does it mean it is safe to remove this?

Reply:
Added comment to indicate that it will be done in the next PR.